src/deepsparse/engine.py

```diff
@@ -804,10 +860,14 @@ def analyze(
             are setup correctly for the DeepSparse Engine
         :return: the analysis structure containing the performance details of each layer
         """
-        if val_inp:
-            self._validate_inputs(inp)
 
-        [out, bench_info] = self._eng_net.benchmark_execute(inp)
+        if self._kv_cache_params:
+            [_, bench_info] = self._eng_net.benchmark_execute(inp, self._kv_cache)
```

Comment:
shouldn't we validate inputs here too?

Reply:
Yeah you right. I forgot to add awareness of kvcache to the validate_inputs(..) method. Added it now and merged the code with a None